packages/core/useScreenOrientation/index.ts

```diff
@@ -0,0 +1,52 @@
+import { ref } from 'vue-demi'
+import { useEventListener } from '../useEventListener'
+import type { ConfigurableWindow } from '../_configurable'
+import { defaultWindow } from '../_configurable'
+
+/**
+ *
+ * Reactive screen orientation
+ *
+ * @see https://vueuse.org/useScreenOrientation
+ */
+export const useScreenOrientation = (options: ConfigurableWindow = {}) => {
+  const {
+    window = defaultWindow,
+  } = options
+
+  const isSupported = !!(window && 'screen' in window && 'orientation' in window.screen)
+
+  const screenOrientation = isSupported ? window.screen.orientation : {} as ScreenOrientation
+
+  const orientation = ref<OrientationType | undefined>(screenOrientation.type)
+  const angle = ref(screenOrientation.angle || 0)
+
+  if (isSupported) {
+    useEventListener(screenOrientation, 'change', () => {
```

Comment:
I guess it would be better to register the listeners this way than on the window.But I tried it and found that the window event fired first, I will modify it back

Reply:
Yeh there are deprecation changes that you needed to be aware of when putting this together.